src/System.Text.Json/tests/Serialization/NullableTests.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Collections.Immutable;
+using System.Linq;
+using System;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class NullableTests
+    {
+        [Fact]
+        public static void DictionaryWithNullableValue()
+        {
+            Dictionary<string, float?> dictWithFloatValue = new Dictionary<string, float?> { { "key", 42.0f } };
```

Comment:
Add `IDictionary` tests too, along with custom implementation of `IDictionary`.

Reply:
IDictionary - https://github.com/dotnet/corefx/blob/5e102fb5d30af4bc940aa519195c0e3dbd933648/src/System.Text.Json/tests/Serialization/NullableTests.cs#L38-L45.

Good point, adding tests for derived classes.